test/replication-luatest/gh_7318_box_wait_limbo_acked_test.lua

```diff
@@ -0,0 +1,545 @@
+
+g.test_update_greatest_term_while_wait_quorum = function(cg)
+    local f = cg.replica:exec(function()
+        box.error.injection.set('ERRINJ_WAL_DELAY_COUNTDOWN', 0)
+        local f = require('fiber').create(function()
+            return pcall(box.ctl.promote)
+        end)
+        f:set_joinable(true)
+        return f:id()
+    end)
+    server_wait_wal_is_blocked(cg.replica)
+    cg.replica_proxy:pause()
+
+    t.helpers.retrying({timeout = wait_timeout}, function()
+        cg.master:exec(function()
+            local status = box.info.replication[2].upstream.status
+            t.assert(status ~= 'follow')
+        end)
+    end)
+
+    cg.master:exec(function()
+        require('fiber').create(function() box.space.test:insert{1} end)
+    end)
+    server_wait_synchro_queue_len_is_equal(cg.replica, 1)
+
+    local csw = get_fiber_csw(cg.replica, f)
+    cg.replica:exec(function()
+        box.error.injection.set('ERRINJ_WAL_DELAY_COUNTDOWN', 0)
+        box.error.injection.set('ERRINJ_WAL_DELAY', false)
+    end)
+    server_wait_fiber_csw_ge_than(cg.replica, f, csw + 1)
+
+    csw = get_fiber_csw(cg.replica, f)
+    local ff = require('fiber').create(function()
+        cg.replica:exec(function(f)
+            box.error.injection.set('ERRINJ_WAL_DELAY', false)
+            local _, ok, err = require('fiber').find(f):join()
+            t.assert(not ok)
+            t.assert_equals(err.type, 'ClientError')
+            local master_id = box.info.replication[1].id
+            t.assert_equals(err.message, 'Instance with replica id '
+                .. master_id .. ' was promoted first')
+        end, {f})
+    end)
+    ff:set_joinable(true)
+    server_wait_fiber_csw_ge_than(cg.replica, f, csw + 1)
+    server_becomes_the_leader_again(cg.master)
+    server_wait_promote_greatest_term_ge_than(cg.replica, 3)
+    cg.replica_proxy:resume()
+    local ok, err = ff:join()
+    t.assert_equals(err, nil)
+    t.assert(ok)
+    server_wait_synchro_queue_len_is_equal(cg.replica, 0)
+end
+
+g.test_txn_limbo_is_empty_while_wait_quorum = function(cg)
+    local f = cg.replica:exec(function()
+        box.error.injection.set('ERRINJ_WAL_DELAY_COUNTDOWN', 0)
+        local f = require('fiber').create(function()
+            return pcall(box.ctl.promote)
+        end)
+        f:set_joinable(true)
+        return f:id()
+    end)
+    server_wait_wal_is_blocked(cg.replica)
+
+    cg.replica_proxy:pause()
+
+    t.helpers.retrying({timeout = wait_timeout}, function()
+        cg.master:exec(function()
+            local status = box.info.replication[2].upstream.status
+            t.assert(status ~= 'follow')
+        end)
+    end)
+
+    cg.master:exec(function()
+        require('fiber').create(function() box.space.test:insert{1} end)
+    end)
+    server_wait_synchro_queue_len_is_equal(cg.replica, 1)
+
+    local repl = cg.replica:exec(function()
+        local repl = table.copy(box.cfg.replication)
+        box.cfg{replication={repl[1]}}
```

Comment:
Can you achieve the same by pausing `master_proxy`?

Reply:
Yes, I really can.